bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -127,6 +127,21 @@ public GarbageCollectorThread(ServerConfiguration conf, LedgerManager ledgerMana
                 Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("GarbageCollectorThread")));
     }
 
+    private EntryLogMetadataMap createEntryLogMetadataMap(boolean gcPersistentEntrylogMetadataMapEnabled) {
+        final String baseDir = this.conf.getGcPersistentEntrylogMetadataMapPath() != null
```

Comment:
reading of getGcPersistentEntrylogMetadataMapPath should be done only if gcPersistentEntrylogMetadataMapEnabled is enabled. so move this line to the 
`if (gcPersistentEntrylogMetadataMapEnabled){..}` block